vega_sim/scenario/benchmark/run.py

```diff
@@ -57,79 +60,94 @@ def _run(
+
+            is_spot = (
+                market.tradable_instrument.instrument.spot != protos.vega.markets.Spot()
+            )
+            is_future = (
+                market.tradable_instrument.instrument.future
+                != protos.vega.markets.Future()
+            )
+            is_perpetual = (
+                market.tradable_instrument.instrument.perpetual
+                != protos.vega.markets.Perpetual()
             )
 
-        asset = service.api.data.get_asset(market_asset)
-        trades = service.api.data.list_trades(
-            market_ids=[market.id],
-            date_range_start_timestamp=market.market_timestamps.pending,
-        )
-        market_data_history = service.api.data.get_market_data_history_by_id(
-            market.id,
-            start_timestamp=market.market_timestamps.pending,
-        )
-        if is_future or is_perpetual:
+            if is_spot:
+                market_asset = market.tradable_instrument.instrument.spot.quote_asset
+            if is_future:
+                market_asset = (
+                    market.tradable_instrument.instrument.future.settlement_asset
+                )
+            if is_perpetual:
+                market_asset = (
+                    market.tradable_instrument.instrument.perpetual.settlement_asset
+                )
+
+            asset = service.api.data.get_asset(market_asset)
+            trades = service.api.data.list_trades(
+                market_ids=[market.id],
+                date_range_start_timestamp=market.market_timestamps.pending,
+            )
+            market_data_history = service.api.data.get_market_data_history_by_id(
+                market.id,
+                start_timestamp=market.market_timestamps.pending,
+            )
             aggregated_balance_history = service.api.data.list_balance_changes(
                 asset_id=asset.id,
                 market_ids=[market.id],
                 party_ids=["network"],
                 account_types=[protos.vega.vega.AccountType.ACCOUNT_TYPE_INSURANCE],
                 date_range_start_timestamp=market.market_timestamps.pending,
             )
-        # Initialise figures
-        fig1 = None
-        fig2 = None
-        fig3 = None
-        # Update figures
-        fig1 = vis.plot.price_monitoring_analysis(market, market_data_history)
-        if is_future or is_perpetual:
-            fig2 = vis.plot.liquidation_analysis(
-                asset, market, trades, market_data_history, aggregated_balance_history
-            )
-        if is_perpetual:
-            funding_periods = service.api.data.list_funding_periods(
-                market_id=market.id,
-                start_timestamp=market.market_timestamps.pending,
-            )
-            funding_period_data_points = (
-                service.api.data.list_funding_period_data_points(
-                    market.id, start_timestamp=market.market_timestamps.pending
+            # Initialise figures
```

Comment:
It doesn't look like we ever actually use fig1/2/3 other than for saving them separately. Should we just do away with the 1/2/3 stuff and immediately create + save if output is true?

Reply:
Good shout thanks, that is much cleaner.